src/operations/execute_operation.ts

```diff
@@ -188,111 +164,113 @@ function executeWithServerSelection<TResult>(
     selector = readPreference;
   }
 
-  const serverSelectionOptions = { session };
-  function retryOperation(originalError: MongoError) {
-    const isWriteOperation = operation.hasAspect(Aspect.WRITE_OPERATION);
-    const isReadOperation = operation.hasAspect(Aspect.READ_OPERATION);
+  const server = await topology.selectServerAsync(selector, { session });
 
-    if (isWriteOperation && originalError.code === MMAPv1_RETRY_WRITES_ERROR_CODE) {
-      return callback(
-        new MongoServerError({
-          message: MMAPv1_RETRY_WRITES_ERROR_MESSAGE,
-          errmsg: MMAPv1_RETRY_WRITES_ERROR_MESSAGE,
-          originalError
-        })
-      );
+  if (session == null || !operation.hasAspect(Aspect.RETRYABLE)) {
```

Comment:
question: under what circumstances would we have a null session?

Reply:
if topology.hasSessionSupport() is false, which is the case if any one of the servers in the topology report a null `logicalSessionTimeoutMinutes`